### ISSUE TITLE
Freetype is returning invalid argument.

### ISSUE THREAD

**Issue by F35H** [sammycage/plutosvg#21]
Not much else to say. I'm using Visual Studio, Windows. Pluto SVG, Luna SVG, and Pluto VG are all statically compiled. This is the define code:

#define PLUTOSVG_HAS_FREETYPE
#define PLUTOSVG_BUILD_STATIC 
#include <plutosvg.h>
#include <ft2build.h>

#include FT_FREETYPE_H
#include FT_MODULE_H

No linking errors, etc. Only error is a 0x06 error code from FT_Set_property. Code is exactly the same as the README.

**Comment by sammycage** [comment]
Thank you for reporting this issue! Are you using CMake for your build process? Also, can you confirm if `PLUTOSVG_HAS_FREETYPE` is defined during compilation? It might also be worth checking if `plutosvg_ft_svg_hooks()` is returning a null pointer.

**Comment by F35H** [comment]
I am not using CMAKE at all.

The Macro doesn't seem to be defined as the function is returning a nullptr.

This issue persists whether I have PLUTOSVG_HAS_FREETYPE defined or not. I tried it simply to see if it'd fix it.

**Comment by sammycage** [comment]
git clone https://github.com/sammycage/plutosvg.git
cd plutosvg
cmake -B build . -DPLUTOSVG_ENABLE_FREETYPE=1
cmake --build build
cmake --install build

**Comment by F35H** [comment]
Update: is not working, but I tried compiling with the define directly in the static library itself - still returns the error.

**Comment by F35H** [comment]
> To enable FreeType integration, you can configure it during the CMake build process like this:
> 
> ```shell
> git clone https://github.com/sammycage/plutosvg.git
> cd plutosvg
> cmake -B build . -DPLUTOSVG_ENABLE_FREETYPE=1
> cmake --build build
> cmake --install build
> ```

Getting this error:
```
CMake Error in CMakeLists.txt:
  Target "plutosvg" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "C:/Users/micha/Downloads/plutosvg-0.0.3/plutosvg-0.0.3/include"

  which is prefixed in the source directory.
```

**Comment by sammycage** [comment]
I've never tried this on Windows, so that error is a bit strange to me. It might be something with how paths are handled on your system. Let me know if you figure it out or need more help!

**Comment by sammycage** [comment]
@F35H How is it going?

**Comment by F35H** [comment]
Only thing different is I tried the CMAKE GUI this time and have the same issue. I will update if anything changes.

**Comment by sammycage** [comment]
What's up? 👋

**Comment by sammycage** [comment]
Hi,

I've updated PlutoSVG so that the FreeType SVG hooks are now located in a dedicated header file, **plutosvg-ft.h**. This change means you no longer need to define `PLUTOSVG_HAS_FREETYPE` during compilation. Separating the FreeType dependency from the core PlutoSVG build can help avoid configuration issues—such as the 0x06 error from `FT_Set_property`—and gives you more control over how you integrate FreeType.

To use the FreeType SVG hooks in your project, update your code as follows:

```cpp
#include <plutosvg-ft.h>

// Configure FreeType to use PlutoSVG's SVG hooks:
FT_Property_Set(library, "ot-svg", "svg-hooks", &plutosvg_ft_hooks);
```

In summary, By moving the SVG hooks into `plutosvg-ft.h`, you can manage the FreeType integration independently from the PlutoSVG build.

Let me know if this update fixes your issue or if you have any further questions!
